apps/builder/app/builder/features/style-panel/sections/transitions/transition-utils.ts

```diff
@@ -63,3 +100,262 @@ export const findTimingFunctionFromValue = (
     (key: TimingFunctions) => timingFunctions[key] === timingFunction
   );
 };
+
+export const transitionProperties = [
+  "transitionProperty",
+  "transitionTimingFunction",
+  "transitionDelay",
+  "transitionDuration",
+] as const;
+
+export type TransitionProperties = (typeof transitionProperties)[number];
+
+export const getTransitionProperties = (
+  currentyStyle: StyleInfo
+): Record<TransitionProperties, LayersValue> => {
+  const properties: Record<TransitionProperties, LayersValue> = {
+    transitionProperty: { type: "layers", value: [] },
+    transitionTimingFunction: { type: "layers", value: [] },
+    transitionDelay: { type: "layers", value: [] },
+    transitionDuration: { type: "layers", value: [] },
+  };
+  for (const property of transitionProperties) {
+    const value = currentyStyle[property];
+    if (value !== undefined && value.value.type === "layers") {
+      properties[property] = value.value;
+    }
+  }
+
+  return properties;
+};
+
+const isValidTransitionValue = (
+  value: LayerValueItem
+): value is KeywordValue => {
+  return value.type === "keyword" || value.type === "unit";
+};
+
+export const convertIndividualTransitionToLayers = (
+  properties: Record<TransitionProperties, LayersValue>
+) => {
+  const layers: { type: "layers"; value: Array<TupleValue> } = {
+    type: "layers",
+    value: [],
+  };
+  const {
+    transitionProperty,
+    transitionDuration,
+    transitionDelay,
+    transitionTimingFunction,
+  } = properties;
+
+  for (const [index] of transitionProperty.value.entries()) {
+    const property = transitionProperty.value[index];
```

Comment:
There is an edge case. Some longhands can have only single value. And this value is used to each transition property.

Reply:
Yes, but here is the case why i am maintaining all the defaults even if there is just one value.
Let's say user is adding these three layers
```
opacity 0ms ease 0ms
width
background 10ms ease-in 10ms 
```

If we saw them as individual properties.

```
transition-property: opacity, width, background
transition-duration: 0ms, 10ms
transition-timing-function: ease, ease-in
transition-delay: 0ms, 10ms
```

Once we read them back to display as individual layers. This will end up coming ass

```
opacity 0ms ease 0ms
width 10ms ease-in 10ms 
background 
```

So, just adding the default values even if there is just one property added by user

Reply:
See what I mean
```
transition-property: opacity, width, background;
transition-duration: 10ms;
```
Here transition should expand into `10ms, 10ms, 10ms`

Reply:
I refactored a little bit, but looks like it's going to be a real tricky one. When users add `transition-property` or any of the long hands using the advanced tab. They get added as `unparsed`. So, we need to handle all these individual cases to make sure they are accessible in the style-panel again.

Will ping once the updates are done 👍